### ISSUE TITLE
Question about Account endpoint implementation

### ISSUE THREAD

**Issue by chriske** [glatzert/ACME-Server-ADCS#4]
First of all, I'd like to thank you for your work, this application is amazing!

I made a PoC integration with your application for one of my clients: I wanted to integrate it with cert-manager (https://cert-manager.io/). If you are not familiar with it, it can request and renew certificates in a Kubernetes environment with ACME protocol. 

I was able to integrate it with your application, but when I wanted to request a new certificate, I found in the IIS logs, that the app throws a NotImplemented exception for the Account endpoints. I'm not familiar with the implementation details of ACME protocol (I don't know that these endpoints are neccessary or not), but I would like to ask: will you implement these endpoints as well in the future?

Thanks for your answer in advance

**Comment by glatzert** [comment]
Hey, I'll check it - the software should be compliant with the ACME-RFC. And I successfully issues certificate already.

But I might not have uploaded the latest build to release.

I'll get around to it this weekend, I'd suppose.


Am 18.08.2022 14:37 schrieb Inotay Krisztián ***@***.***>:

First of all, I'd like to thank you for your work, this application is amazing!

I made a PoC integration with your application for one of my clients: I wanted to integrate it with cert-manager (https://cert-manager.io/). If you are not familiar with it, it can request and renew certificates in a Kubernetes environment with ACME protocol.

I was able to integrate it with your application, but when I wanted to request a new certificate, I found in the IIS logs, that the app throws a NotImplemented exception for the Account endpoints. I'm not familiar with the implementation details of ACME protocol (I don't know that these endpoints are neccessary or not), but I would like to ask: will you implement these endpoints as well in the future?

Thanks for your answer in advance

**Comment by chriske** [comment]
Hey, thanks for the response!

These are the methods what I'm talking about:

https://github.com/glatzert/ACME-Server-ACDS/blob/8b459f8b5a06fec15089a5b0eb61705a5e8da547/src/ACME.Server/Controllers/AccountController.cs#L52

**Comment by glatzert** [comment]
To be fair, I never used that function.
Nevertheless, they’ll get implemented :)

I wonder why your tool uses them …

Thomas Ottenhus
Zentrum für Datenverarbeitung (ZDV)
Leiter Softwareentwicklung

Johannes Gutenberg-Universität Mainz
Anselm-Franz-von-Bentzel-Weg 12, 55128 Mainz

Tel: +49 6131 39 26487
***@***.******@***.***>
***@***.******@***.***>
https://www.zdv.uni-mainz.de<https://www.zdv.uni-mainz.de/>

**Comment by glatzert** [comment]
I took a first look into the "FindAccount" functionality.
It should not be triggered, if the tool does not submit "existingAccountOnly" (a parameter controlling, if you create or look for an account). Nevertheless it might do that before creation to not run into an error, if the account already exists (but into an error, if it does not).

The implementation of that functionality is a little bit more work than anticipated, which esentially is my fault, since I did not use SQLite for storage, but the harddrive AND decided to use a folder named by the accountId.

I'll have to do a little rewriting to fix that problem, so I'll need another week.
